aliyun-ecs/src/main/java/org/jclouds/aliyun/ecs/features/SecurityGroupApi.java

```diff
@@ -0,0 +1,141 @@
+@RequestFilters(FormSign.class)
+@QueryParams(keys = { "Version", "Format", "SignatureVersion", "ServiceCode", "SignatureMethod" },
+             values = {"{" + Constants.PROPERTY_API_VERSION + "}", "JSON", "1.0", "ecs", "HMAC-SHA1"})
+public interface SecurityGroupApi {
+
+   @Named("securityGroup:list")
+   @GET
+   @QueryParams(keys = "Action", values = "DescribeSecurityGroups")
+   @ResponseParser(ParseSecurityGroups.class)
+   @Fallback(Fallbacks.EmptyIterableWithMarkerOnNotFoundOr404.class)
+   IterableWithMarker<SecurityGroup> list(@QueryParam("RegionId") String region, ListSecurityGroupsOptions options);
+
+   @Named("securityGroup:list")
+   @GET
+   @QueryParams(keys = "Action", values = "DescribeSecurityGroups")
+   @ResponseParser(ParseSecurityGroups.class)
+   @Transform(ParseSecurityGroups.ToPagedIterable.class)
+   @Fallback(Fallbacks.EmptyPagedIterableOnNotFoundOr404.class)
+   PagedIterable<SecurityGroup> list(@QueryParam("RegionId") String region);
+
+   @Singleton
+   final class ParseSecurityGroups extends ParseJson<SecurityGroups> {
+
+      @Inject
+      ParseSecurityGroups(final Json json) {
+         super(json, TypeLiteral.get(SecurityGroups.class));
+      }
+
+      static class ToPagedIterable extends Arg0ToPagedIterable<SecurityGroup, ParseSecurityGroups.ToPagedIterable> {
+
+         private final ECSComputeServiceApi api;
+
+         @Inject
+         ToPagedIterable(ECSComputeServiceApi api) {
+            this.api = api;
+         }
+
+         @Override
+         protected Function<Object, IterableWithMarker<SecurityGroup>> markerToNextForArg0(final Optional<Object> arg0) {
+            return new Function<Object, IterableWithMarker<SecurityGroup>>() {
+               @Override
+               public IterableWithMarker<SecurityGroup> apply(Object input) {
+                  String regionId = arg0.get().toString();
+                  ListSecurityGroupsOptions listSecurityGroupsOptions = ListSecurityGroupsOptions.Builder.paginationOptions(PaginationOptions.class.cast(input));
+                  return api.securityGroupApi().list(regionId, listSecurityGroupsOptions);
+               }
+            };
+         }
+      }
+   }
+
+   @Named("securityGroup:get")
+   @GET
+   @QueryParams(keys = "Action", values = "DescribeSecurityGroupAttribute")
+   @SelectJson("Permission")
+   List<Permission> get(@QueryParam("RegionId") String region, @QueryParam("SecurityGroupId") String securityGroupId);
+
+   @Named("securityGroup:create")
+   @POST
+   @QueryParams(keys = "Action", values = "CreateSecurityGroup")
+   @Fallback(Fallbacks.EmptyListOnNotFoundOr404.class)
+   SecurityGroupRequest create(@QueryParam("RegionId") String region);
+
+   @Named("securityGroup:create")
+   @POST
+   @QueryParams(keys = "Action", values = "CreateSecurityGroup")
+   @Fallback(Fallbacks.EmptyListOnNotFoundOr404.class)
+   SecurityGroupRequest create(@QueryParam("RegionId") String region, CreateSecurityGroupOptions options);
```

Comment:
Options will be bound to the request as query params. Is that correct for this POST operation?